src/main/public/js/postcode-lookup.js

```diff
@@ -0,0 +1,158 @@
+        .addEventListener('change', function (event) {
+          var addressDetails = this.value.split(', ')
+          var addressSelector = postcodeLookupWidget.querySelector('.address')
+          addressSelector.querySelector('.address-line1').value = addressDetails[0]
+          addressSelector.querySelector('.address-line2').value = addressDetails[1]
+          addressSelector.querySelector('.address-town-or-city').value = addressDetails[2]
+          addressSelector.querySelector('.postcode').value = addressDetails[3]
+          showAddressEntry(postcodeLookupWidget)
+        })
+
+      if (isAnyAddressFieldPopulated(postcodeLookupWidget)) {
+        showAddressEntry(postcodeLookupWidget)
+      }
+    })
+
+  document.querySelector('#saveAndContinue')
+    .addEventListener('click', function (event) {
+      document.querySelectorAll('.postcode-lookup')
+        .forEach(function(postcodeLookupWidget) {
+          var addressInputsVisible = !postcodeLookupWidget.querySelector('.address')
+            .classList.contains('js-hidden')
+          console.log('address', addressInputsVisible)
+          postcodeLookupWidget.querySelector('.postcode-address-visible').value = addressInputsVisible
+
+
+          var addressPickerVisible = !postcodeLookupWidget.querySelector('.postcode-address-picker')
+            .classList.contains('hidden')
+          console.log('picker', addressPickerVisible)
+          postcodeLookupWidget.querySelector('.address-selector-visible').value = addressPickerVisible
+        })
+    })
+
+})
+
+function isNorthernIrelandPostcode (postcode) {
+  return postcode.startsWith('BT')
+}
+
+function showAddressEntry (postcodeLookupWidget) {
+  postcodeLookupWidget.querySelector('.address').classList.remove('js-hidden')
+}
+
+function clearPostcodeDropdown (postcodeLookupWidget) {
+  postcodeLookupWidget.querySelector('select').innerHTML = ''
+}
+
+function hidePostcodeDropdown (postcodeLookupWidget) {
+  postcodeLookupWidget.querySelector('.postcode-address-picker').classList.add('hidden')
+}
+
+function showPostcodeDropdown (postcodeLookupWidget) {
+  postcodeLookupWidget.querySelector('.postcode-address-picker').classList.remove('hidden')
+}
+
+function clearAddressFields (postcodeLookupWidget) {
+  var addressSelector = postcodeLookupWidget.querySelector('.address')
+  addressSelector.querySelector('.address-line1').value = ''
+  addressSelector.querySelector('.address-line2').value = ''
+  addressSelector.querySelector('.address-town-or-city').value = ''
+  addressSelector.querySelector('.postcode').value = ''
+}
+
+function isAnyAddressFieldPopulated(postcodeLookupWidget) {
+  var addressSelector = postcodeLookupWidget.querySelector('.address')
+
+  return addressSelector.querySelector('.address-line1').value !== '' ||
+    addressSelector.querySelector('.address-line2').value !== '' ||
+    addressSelector.querySelector('.address-town-or-city').value !== '' ||
```

Comment:
I can see plenty of `addressSelector.querySelector` calls, some of them to the same DOM element. What do you think about extracting at least some of them and then have e.g. `addressLine2().value`?

Reply:
If they are always in DOM I would even assign reference to DOM node and use var:
```
var element = addressSelector.querySelector('.address-line2')
element.value = ...
```

Reply:
@yarpo I don't think any of them are in scope for that to work and give value,
I'll have a look at what rafal suggested after I have the tests fixed